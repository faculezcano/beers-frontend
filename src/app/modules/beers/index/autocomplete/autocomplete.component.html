<ng-autocomplete
  [data]="list"
  searchKeyword="value"
  (inputChanged)="input($event)"
  (selected)="handleSelected($event)"
  [itemTemplate]="itemTemp"
>
</ng-autocomplete>

<ng-template #itemTemp let-result>
  <div class="row gx-0 text-start">
    <div class="col p-2" [innerHTML]="result.value"></div>
    <div class="col-auto p-2 text-muted small d-flex align-items-end">
      {{ result.type }}
    </div>
  </div>
</ng-template>
