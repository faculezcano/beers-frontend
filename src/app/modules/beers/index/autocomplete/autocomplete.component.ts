import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { BeersService } from '../../../../services/beers.service';

@Component({
  selector: 'app-autocomplete',
  templateUrl: './autocomplete.component.html',
  styleUrls: ['./autocomplete.component.scss'],
})
export class AutocompleteComponent implements OnInit {
  @Input() url: string = '';
  @Output() selected = new EventEmitter<any>();

  public list: any[] = [];
  public value: string = '';

  private searchTimer: number | undefined;

  constructor(private beersService: BeersService) {}

  ngOnInit(): void {}

  input(term: string) {
    this.list = [];

    if (this.searchTimer) {
      clearTimeout(this.searchTimer);
    }

    if (term) {
      this.searchTimer = setTimeout(() => {
        this.search(term);
      }, 500);
    }
  }

  search(term: string) {
    this.beersService.search(term).subscribe((results) => {
      this.list = (results as any[]) || [];
    });
  }

  handleSelected(event: any) {
    this.selected.emit(event);
    this.value = event.value;
  }
}
