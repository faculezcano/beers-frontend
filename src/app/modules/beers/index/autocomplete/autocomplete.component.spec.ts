import { ComponentFixture, TestBed } from '@angular/core/testing';

import { AutocompleteComponent } from './autocomplete.component';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { AutocompleteLibModule } from 'angular-ng-autocomplete';

describe('AutocompleteComponent', () => {
  let component: AutocompleteComponent;
  let fixture: ComponentFixture<AutocompleteComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [AutocompleteComponent],
      imports: [HttpClientTestingModule, AutocompleteLibModule],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AutocompleteComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
