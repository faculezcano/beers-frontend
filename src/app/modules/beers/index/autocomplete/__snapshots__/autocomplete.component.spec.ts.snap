// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteComponent should match snapshot 1`] = `
<app-autocomplete
  __ngContext__={[Function LRootView]}
  beersService={[Function BeersService]}
  list={[Function Array]}
  selected={[Function EventEmitter_]}
  url=""
  value=""
>
  <ng-autocomplete
    class="ng-autocomplete"
    searchkeyword="value"
  >
    <div
      aria-expanded="false"
      aria-owns="suggestions suggestions-history"
      class="autocomplete-container"
    >
      <div
        class="input-container"
      >
        <input
          aria-autocomplete="list"
          aria-label=""
          autocomplete="off"
          class="ng-untouched ng-pristine ng-valid"
          placeholder=""
          role="combobox"
          type="text"
        />
      </div>
      <div
        class="suggestions-container is-hidden"
        id="suggestions"
        role="listbox"
      >
        <ul>
        </ul>
      </div>
      <div
        class="suggestions-container is-visible"
        id="suggestions-history"
        role="listbox"
      >
        <ul>
        </ul>
      </div>
    </div>
  </ng-autocomplete>
</app-autocomplete>
`;
